core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -225,10 +225,24 @@ abstract class RDD[T: ClassTag](
   /** Get the RDD's current storage level, or StorageLevel.NONE if none is set. */
   def getStorageLevel: StorageLevel = storageLevel
 
+  /**
+   * Lock for all mutable state of this RDD (persistence, partitions, dependencies, etc.).  We do
+   * not use `this` because RDDs are user-visible, so users might have added their own locking on
+   * RDDs; sharing that could lead to a deadlock.
+   *
+   * One thread might hold the lock on many of these, for a chain of RDD dependencies; but
+   * because DAGs are acyclic, and we only ever hold locks for one path in that DAG, there is no
+   * chance of deadlock.
+   *
+   * The use of Integer is simply so this is serializable -- executors may reference the shared
+   * fields (though they should never mutate them, that only happens on the driver).
+   */
+  private val stateLock = new Integer(0)
```

Comment:
The Integer constructor has been deprecated already, see https://docs.oracle.com/javase/9/docs/api/java/lang/Integer.html . This yields the warning:
```
RDD.scala:240: constructor Integer in class Integer is deprecated: see corresponding Javadoc for more information.
``` 
Is it possible to replace it by something else?

Reply:
I tried to eliminate the warning in https://github.com/apache/spark/pull/27399